core/templates/dev/head/services/context.service.ts

```diff
@@ -17,193 +17,207 @@
+          return ServicesConstants.PAGE_CONTEXT.EXPLORATION_PLAYER;
+        } else if (pathnameArray[i] === 'create') {
+          this.pageContext = ServicesConstants.PAGE_CONTEXT.EXPLORATION_EDITOR;
+          return ServicesConstants.PAGE_CONTEXT.EXPLORATION_EDITOR;
+        } else if (pathnameArray[i] === 'question_editor') {
+          this.pageContext = ServicesConstants.PAGE_CONTEXT.QUESTION_EDITOR;
+          return ServicesConstants.PAGE_CONTEXT.QUESTION_EDITOR;
+        } else if (pathnameArray[i] === 'topic_editor') {
+          this.pageContext = ServicesConstants.PAGE_CONTEXT.TOPIC_EDITOR;
+          return ServicesConstants.PAGE_CONTEXT.TOPIC_EDITOR;
+        } else if (pathnameArray[i] === 'story_editor') {
+          this.pageContext = ServicesConstants.PAGE_CONTEXT.STORY_EDITOR;
+          return ServicesConstants.PAGE_CONTEXT.STORY_EDITOR;
+        } else if (pathnameArray[i] === 'skill_editor') {
+          this.pageContext = ServicesConstants.PAGE_CONTEXT.SKILL_EDITOR;
+          return ServicesConstants.PAGE_CONTEXT.SKILL_EDITOR;
+        } else if (
+          pathnameArray[i] === 'practice_session' ||
+            pathnameArray[i] === 'review_test') {
+          this.pageContext = ServicesConstants.PAGE_CONTEXT.QUESTION_PLAYER;
+          return ServicesConstants.PAGE_CONTEXT.QUESTION_PLAYER;
+        } else if (pathnameArray[i] === 'collection_editor') {
+          this.pageContext = ServicesConstants.PAGE_CONTEXT.COLLECTION_EDITOR;
+          return ServicesConstants.PAGE_CONTEXT.COLLECTION_EDITOR;
         }
-        return decodeURI(pathnameArray[2]);
-      },
+      }
 
-      getEntityType: function() {
-        var pathnameArray = UrlService.getPathname().split('/');
-        for (var i = 0; i < pathnameArray.length; i++) {
-          if (pathnameArray[i] === 'create' || pathnameArray[i] === 'explore' ||
-            (pathnameArray[i] === 'embed' &&
-             pathnameArray[i + 1] === 'exploration')) {
-            return ENTITY_TYPE.EXPLORATION;
-          }
-          if (pathnameArray[i] === 'topic_editor') {
-            return ENTITY_TYPE.TOPIC;
-          }
-          if (pathnameArray[i] === 'subtopic') {
-            return ENTITY_TYPE.SUBTOPIC;
-          }
-          if (pathnameArray[i] === 'story_editor') {
-            return ENTITY_TYPE.STORY;
-          }
-          if (pathnameArray[i] === 'skill_editor') {
-            return ENTITY_TYPE.SKILL;
-          }
+      return ServicesConstants.PAGE_CONTEXT.OTHER;
+    }
+  }
+
+  isInExplorationContext(): boolean {
+    return (this.getPageContext() ===
+        ServicesConstants.PAGE_CONTEXT.EXPLORATION_EDITOR ||
+        this.getPageContext() ===
+        ServicesConstants.PAGE_CONTEXT.EXPLORATION_PLAYER);
+  }
+
+  isInQuestionContext(): boolean {
+    return (
+      this.getPageContext() === ServicesConstants.PAGE_CONTEXT.QUESTION_EDITOR);
+  }
+
+  getEntityId(): string {
```

Comment:
This function and `getEntityType()` were modified in develop. Also, some of the 'question' functions were also removed. These are the merge conflicts.

Reply:
@aks681 Okay thanks for the heads up! I'll resolve those and update my PR! Thanks!

Reply:
Done